modules/oe_whitelabel_paragraphs/tests/src/Functional/InstallTest.php

```diff
@@ -0,0 +1,203 @@
+      'field_oe_text' => 'More information',
+      // These fields will be renamed.
+      'oe_bt_links_block_orientation' => 'vertical',
+      'oe_bt_links_block_background' => 'gray',
+      'field_oe_links' => [
+        // One link is enough for this test.
+        [
+          'title' => 'European Commission',
+          'uri' => 'https://example.com',
+        ],
+      ],
+    ];
+    $paragraphs_data['oe_social_media_follow'] = [
+      'type' => 'oe_social_media_follow',
+      'field_oe_title' => 'Social media title',
+      'field_oe_social_media_variant' => 'horizontal',
+      // This field will be renamed.
+      'oe_bt_links_block_background' => 'gray',
+      // One link is enough for this test.
+      'field_oe_social_media_links' => [
+        [
+          'title' => 'Email',
+          'uri' => 'mailto:example@com',
+          'link_type' => 'email',
+        ],
+      ],
+      'field_oe_social_media_see_more' => [
+        'title' => 'Other social networks',
+        'uri' => 'https://europa.eu/european-union/contact/social-networks_en',
+      ],
+    ];
+
+    $ids = [];
+    foreach ($paragraphs_data as $name => $paragraph_data) {
+      $paragraph = Paragraph::create($paragraph_data);
+      $paragraph->save();
+      $ids[$name] = $paragraph->id();
+    }
+
+    self::assertTrue($this->moduleHandler->moduleExists('oe_whitelabel_legacy_paragraphs_test'));
+    self::assertFalse($this->moduleHandler->moduleExists('oe_whitelabel_paragraphs'));
+    self::assertTrue($this->moduleInstaller->install(['oe_whitelabel_paragraphs']));
+    self::assertFalse($this->moduleHandler->moduleExists('oe_whitelabel_paragraphs'));
+    $this->reloadServices();
+    self::assertTrue($this->moduleHandler->moduleExists('oe_whitelabel_paragraphs'));
+
+    $expected_created = [
+      'oe_description_list' => [
+        'oe_w_orientation' => 'horizontal',
+      ],
+      'oe_facts_figures' => [
+        'oe_w_n_columns' => '3',
+      ],
+      'oe_links_block' => [
+        'oe_w_links_block_orientation' => 'vertical',
+        'oe_w_links_block_background' => 'gray',
+      ],
+      'oe_social_media_follow' => [
+        'oe_w_links_block_background' => 'gray',
+      ],
+    ];
+
+    $expected_deleted = [
+      'oe_description_list' => [
+        'oe_bt_orientation' => TRUE,
+      ],
+      'oe_facts_figures' => [
+        'oe_bt_n_columns' => TRUE,
+      ],
+      'oe_links_block' => [
+        'oe_bt_links_block_orientation' => TRUE,
+        'oe_bt_links_block_background' => TRUE,
+      ],
+      'oe_social_media_follow' => [
+        'oe_bt_links_block_background' => TRUE,
+      ],
+    ];
+
+    $actual_updated = [];
+    $actual_deleted = [];
+    foreach ($ids as $name => $id) {
+      $updated_paragraph = Paragraph::load($id);
+      self::assertNotNull($updated_paragraph);
+      foreach ($expected_created[$name] as $field_name => $value) {
+        if (!$updated_paragraph->hasField($field_name)) {
```

Comment:
this is a bit confusing to me, so I think a good candidate for adding a comment above the if, explaining why we continue there.

Reply:
Fyi, I prefer to put comment inside the `if ()`, because it eliminates the need for conditional phrases in the comment.
I don't know of any convention that would prevent this, and it is generally better.

```php
// Skip if node not found.
if ($node === NULL) {
  return;
}
```
vs
```php
if ($node === NULL) {
  // Node was not found. Skip.
  return;
}
```